src/core/etl/src/Flow/Serializer/NativePHPSerializer.php

```diff
@@ -17,14 +17,18 @@ public function serialize(Serializable $serializable) : string
         return \serialize($serializable);
     }
 
-    public function unserialize(string $serialized) : Serializable
+    public function unserialize(string $serialized, string $class) : Serializable
     {
         $value = \unserialize($serialized, ['allowed_classes' => true]);
 
         if (!$value instanceof Serializable) {
             throw new RuntimeException('NativePHPSerializer::unserialize must return instance of Serializable');
         }
 
+        if (!\is_a($value, $class)) {
```

Comment:
I think you might want to pass class name to allowed_class option and then simplify that condition.